### PR TITLE
Add multi fields support

### PR DESCRIPTION
**Pull request by chris-asl**
* Add support for querying multiple fields. This is achieved by either
   using the `multi_match` query (simple value case), or `query_string`
   by employing the `fields` field, or with multiple `should` `bool`
   queries for the case of exact matches.

### CODE REVIEW

File: tests/test_elastic_search_visitor.py [hunk 1]
```diff
@@ -459,23 +458,108 @@ def test_elastic_search_visitor_with_query_with_malformed_part_and_default_malfo
...
+                "fields": ["earliest_date", "preprint_date"],
+                "query": "2000-10-*",
+            }
+        }
+
+    generated_es_query = _parse_query(query_str)
+    assert generated_es_query == expected_es_query
+
+
+def test_elastic_search_visitor_with_multi_match_when_es_field_is_a_list_and_exact_match_value():
```
Comment (chris-asl) [inline tests/test_elastic_search_visitor.py]: @jacquerie Can you check whether this implementation for multiple fields for an `exact` query looks ok?

File: inspire_query_parser/visitors/elastic_search_visitor.py [hunk 2]
```diff
@@ -45,6 +45,18 @@ class ElasticSearchVisitor(Visitor):
     Notes:
         The ElasticSearch query follows the 2.4 version DSL specification.
     """
+    def _generate_query_string_query(self, value, fieldnames, analyze_wildcard):
+        field_specifier, field_specifier_value = ('default_field', '_all') if not fieldnames \
+            else ('fields', fieldnames if isinstance(fieldnames, list) else [fieldnames])
```
Comment (jacquerie) [inline inspire_query_parser/visitors/elastic_search_visitor.py]: I'd say that there's too much logic on one line.

File: inspire_query_parser/visitors/elastic_search_visitor.py [hunk 3]
```diff
@@ -175,9 +180,9 @@ def visit_keyword(self, node):
             'author': 'authors.full_name',
             'citedby': 'citedby',
             'collaboration': 'collaborations.value',
-            'date': 'earliest_date',
+            'date': ['earliest_date', 'preprint_date'],
             'doi': 'dois.value.raw',
-            'eprint': 'arxiv_eprints.value.raw',
+            'eprint': 'arxiv_eprints.value',
```
Comment (jacquerie) [inline inspire_query_parser/visitors/elastic_search_visitor.py]: Why this change?

File: tests/test_elastic_search_visitor.py [hunk 4]
```diff
@@ -459,23 +458,108 @@ def test_elastic_search_visitor_with_query_with_malformed_part_and_default_malfo
...
+                    }
+                ]
+            }
+        }
+
+    generated_es_query = _parse_query(query_str)
+    assert generated_es_query == expected_es_query
+
+
+def test_elastic_search_visitor_with_multi_match_when_es_field_is_a_list_and_exact_match_value_with_wildcard():
```
Comment (jacquerie) [inline tests/test_elastic_search_visitor.py]: Something's wrong with this test? It looks identical to the one above, minus the name.

File: inspire_query_parser/visitors/elastic_search_visitor.py (hunk 2 above)
Comment (chris-asl) [reply inspire_query_parser/visitors/elastic_search_visitor.py]: I agree, will make that clearer!

File: tests/test_elastic_search_visitor.py (hunk 4 above)
Comment (chris-asl) [reply tests/test_elastic_search_visitor.py]: Oops, that shouldn't make it online! Will remove it.

File: inspire_query_parser/visitors/elastic_search_visitor.py (hunk 3 above)
Comment (chris-asl) [reply inspire_query_parser/visitors/elastic_search_visitor.py]: Hmm, this is a bit weird: 
Last week @StellaCh reported that `find eprint 0803.0492` is not working properly as it was returing many results.
I tried that by querying labs and could reproduce the "bug".  
So, I changed the query from `find eprint 0803.0492` to `find arxiv_eprints.value: 0803.0492` and it worked fine. 

Today, I tried querying again after the deployment on labs and couldn't reproduce it (i.e. `eprint 0803.0492` works as expected).  
So, I will change it back.

**Comment by chris-asl** [comment]
@jacquerie regarding the implementation of an exact query with multiple fields?

File: tests/test_elastic_search_visitor.py (hunk 1 above)
Comment (jacquerie) [reply tests/test_elastic_search_visitor.py]: I think it does, but what does ES think? : )

Comment (chris-asl) [reply tests/test_elastic_search_visitor.py]: It does work. For example, this query returned [one hit](http://localhost:5000/api/literature/1498589)
```
{
   "query": {
      "bool": {
          "should": [
              {
                "term": {
                    "earliest_date": "2016-11-16"
                }
              },
              {
                 "term": {
                    "preprint_date": "2016-11-16"
                }
              }
          ]
      }
   }
}
```.

I also tried with `earliest_date` and `legacy_creation_date` whose value was different than `2016-11-16` and still got the same, correct hit.
